internal/provider/resource_tsuru_app_grant.go

```diff
@@ -53,7 +54,11 @@ func resourceTsuruApplicationGrantCreate(ctx context.Context, d *schema.Resource
 	team := d.Get("team").(string)
 
 	err := resource.RetryContext(ctx, d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
-		_, err := provider.TsuruClient.AppApi.AppTeamGrant(ctx, app, team)
+		response, err := provider.TsuruClient.AppApi.AppTeamGrant(ctx, app, team)
+		// ignore teams already granted for this app
+		if response.StatusCode == http.StatusConflict {
+			return nil
+		}
```

Comment:
I'd combined this validation with an extra one against the message error which Tsuru returns - "team already have access to this app". A simple `strings.Contains` in the error message should cover it.

Reply:
Thanks for the tips!
What would happen if the tsuru's upstream API changes the error message? Would it break?

Reply:
Probably... I agree, it's a tradeoff. I took a look on Tsuru code and realized that it only returns 409 in this condition (https://github.com/tsuru/tsuru/blob/main/api/app.go#L869)... so I think it's safe go for it.